### PR TITLE
generate RPC docs using Slate

### LINKED ISSUE #691: Add RPC documentation
Let's add RPC documentation in this style.
https://h.readthedocs.io/en/latest/api/#operation/root

It looks super sexy and is easily usable. The sources for it can be found [here](https://github.com/hypothesis/h/tree/master/docs/_extra/api).

### PR DESCRIPTION
**Pull request by zramsay**
- closes #691 
- the flow is pretty much as described in my last comment on that issue: 
- The Tendermint CI is responsible for checking if any rpc/core files were changed, running the `godoc2md` converter, assembling the file, then committing and pushing the changes to https://github.com/tendermint/slate (that's basically the `scripts/slate.sh` file). Pushing to the master branch is fine in this case since this repo has no active development. The commit triggers the [Slate CI](https://github.com/tendermint/slate/pull/2) to run the `./deploy.sh` script, which pushes the required a new commit to the `gh-pages` branch, which GH graciously serves for us at: https://tendermint.github.io/slate.

Still TODO:
- [x] implement file check (i.e., don't run if nothing in `rpc/core` has changed)
- [x] implement branch restrictions (only run this CI target on merge to develop)

Shout out to @xla for behind the scenes help debugging

### CODE REVIEW

File: docs/specification/rpc.rst [hunk 1]
```diff
@@ -1,190 +1,4 @@
 RPC
 ===
 
-Coming soon: RPC docs powered by `slate <https://github.com/lord/slate>`__. Until then, read on.
-
-Tendermint supports the following RPC protocols:
```
Comment (ebuchman) [inline docs/specification/rpc.rst]: cool, this content is now in `/rpc/core/doc.go`

File: scripts/slate.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+if [ "$CIRCLE_BRANCH" != "develop" ]; then
+	echo "the branch being built isn't develop, exiting"
+	exit 0
+else
+	echo "on develop, building the RPC docs"
+fi
+
+# godoc2md used to convert the go documentation from
+# `rpc/core` into a markdown file consumed by Slate
+go get github.com/melekes/godoc2md
```
Comment (ebuchman) [inline scripts/slate.sh]: I think we can use the upstream one again: https://github.com/davecheney/godoc2md

anton's changes got merged in.

**Comment by zramsay** [comment]
passing: https://circleci.com/gh/tendermint/tendermint/9608
if the branch isn't develop, the test exits with 0, so here's the output when rpc/core is changed but the branch isn't develop:
```
changes detected in rpc/core, continuing
the branch being built isn't develop, exiting
```

I'll see whats up with the links (they're normal md links) but let's get this merged in so we ensure the rest of the automation is smooth

**Comment by ebuchman** [comment]
> if the branch is develop and there are changes, the CI will commit and push to master on https://github.com/tendermint/slate

Wait, so we only get one version of the slate docs?  I dont think we want to push develop changes live if this is our official docs site for the RPC. Ideally we'd have two, master and develop, but if we can only have one, it should be master, and devs can just build it locally to see what develop docs look like.

**Comment by zramsay** [comment]
there's an open issue for multiple versions: https://github.com/lord/slate/issues/138

I'll switch to build on merges to master

File: rpc/core/README.md [hunk 3]
```diff
@@ -3,17 +3,16 @@
...
 
 We are using [Slate](https://github.com/tendermint/slate) to power our RPC
-documentation. If you are changing a comment, make sure to copy the resulting
-changes to the slate repo and make a PR
-[there](https://github.com/tendermint/slate) as well. For generating markdown
-use:
+documentation. For generating markdown use:
 
 ```shell
 go get github.com/melekes/godoc2md
```
Comment (melekes) [inline rpc/core/README.md]: `go get github.com/davecheney/godoc2md`

**Pull request merged**